docs/authentication.md

```diff
@@ -0,0 +1,27 @@
+# Authentication
+
+## Devise
+
+We use [Devise](https://github.com/heartcombo/devise) as our primary authentication solution.
+
+### Configuration
+
+In the provided codebase, devise in configured in `config/initializers/devise.rb`.
+
+1. If using the Rails mailer, don't forget to set the value of the constant `DEFAULT_FROM_EMAIL` for your application.
+
+2. If modifying anything related to the sign up or confirmation process, refer to the controllers `RegistrationsController` and `ConfirmationsController` which override and/or inherits from their original devise controllers.
```

Comment:
this can be a paragraph since the above line won't be there. say that we have made some modifications on how the default registration and confirmation controllers work and why. 

so mention about the exact changes we have made. the confirm email page after sign up, the delete account flow, etc. I think there are more controllers to add to the list. 

we are looking to communicate exactly what we have added on top of default rails. we can assume that anyone using RapidRails knows Rails and how the general gems work. They just want to know what we do differently here in our template here. 

They should know reading our docs here on what we do instead of them having to look in the codebase to find out how it works.